assets/javascripts/discourse/initializers/preview-edits.js.es6

```diff
@@ -160,6 +160,15 @@ export default {
           this.$('.topic-intro').prependTo(this.$('.main-link'))
           this.$('.topic-title').prependTo(this.$('.main-link'))
         }
+
+        if (socialMediaStyle) {
```

Comment:
@pmusaraj Did you try injecting the router into the component? I haven't tried something like this recently, but it may work.